spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java

```diff
@@ -166,8 +165,11 @@ public TransactionInfo getTransactionInfo() {
     }
 
     private <T extends Annotation> T getAnnotation(Method method, Class<?> targetClass, Class<T> annotationClass) {
-        return method == null ? targetClass == null ? null : targetClass.getAnnotation(annotationClass)
-            : method.getAnnotation(annotationClass);
+        T annotation = null == method ? null : method.getAnnotation(annotationClass);
+        if (null == annotation) {
+            annotation = null == targetClass ? null : targetClass.getAnnotation(annotationClass);
+        }
+        return annotation;
     }
```

Comment:
How about to use lambda for more readable like :

```java
        return Optional.ofNullable(method)
                       .map(m -> m.getAnnotation(annotationClass))
                       .orElse(Optional.ofNullable(targetClass)
                                       .map(t -> t.getAnnotation(annotationClass))
                                       .orElse(null));
```

Reply:
thx